packages/jest-snapshot/src/State.ts

```diff
@@ -104,7 +105,9 @@ export default class SnapshotState {
     this._dirty = true;
     if (options.isInline) {
       const error = options.error || new Error();
-      const lines = getStackTraceLines(error.stack || '');
+      const lines = getStackTraceLines(
+        removeLinesBeforeExternalMatcherTrap(error.stack || ''),
```

Comment:
Should `getTopFrame` do this work internally instead of the caller having to do it?

Reply:
I think the stack trace containing `__EXTERNAL_MATCHER_TRAP__` will already got removed by `getStackTraceLines`?

Reply:
Should we instead modify `getStackTraceLines`? Or both?

Reply:
hmm, good point. Is it enough to just change https://github.com/facebook/jest/blob/9ac2dcd55c0204960285498c590c1aa7860e6aa8/packages/jest-message-util/src/index.ts#L49 to not filter out this new capture line? If not I think the approach you have here makes sense

Reply:
I tried, it works find in this context, but it would break in `formatStackTrace`.

https://github.com/facebook/jest/blob/9ac2dcd55c0204960285498c590c1aa7860e6aa8/packages/jest-message-util/src/index.ts#L254-L255

If we omit all stack traces before the trap, then the custom matcher won't correctly format the stack trace. It's covered here in an e2e test.

https://github.com/facebook/jest/blob/9ac2dcd55c0204960285498c590c1aa7860e6aa8/e2e/custom-matcher-stack-trace/__tests__/sync.test.js#L41-L54

I believe that we have 2 options:

1. Leave it be like this to only remove stack traces from the caller
2. Add logic in `formatStackTrace` to **un-skip** the trap.

I think (1) is more resilient while (2) feel more error-prone. Wonder what's your opinion on this?